libsolidity/formal/Z3Interface.cpp

```diff
@@ -25,12 +25,13 @@ using namespace dev;
 using namespace dev::solidity::smt;
 
 Z3Interface::Z3Interface():
-	m_solver(m_context)
+	m_context(make_shared<z3::context>()),
```

Comment:
Z3 doesn't like having two contexts alive, so in case we're using Z3CHCInterface, which uses Z3Interface for variables, the context needs to be the same.